contrib_bots/lib/xkcd.py

```diff
@@ -0,0 +1,134 @@
+from random import randint
+import logging
+import re
+import requests
+
+XKCD_TEMPLATE_URL = 'http://xkcd.com/%s/info.0.json'
+LATEST_XKCD_URL = 'http://xkcd.com/info.0.json'
+
+
+class XkcdHandler(object):
+    '''
+    This plugin provides several command that can be used for fetch comics from
+    xkcd. The bot looks for messages starting with "@xkcd" and responds with a
+    message with the comic based on provided commands.
+    '''
+
+    def usage(self):
+        return '''
+            This plugin allows users to fetch comics provided by xkcd.
+            Users should preface the command with "@xkcd".
+
+            There are several commands to use this bot:
+            - @xkcd help -> To show all commands the bot supports.
+            - @xkcd latest -> To fetch latest comic from xkcd.
+            - @xkcd random -> To fetch random comic from xkcd.
+            - @xkcd <comic_id> -> To fetch comic based on `<comic_id>`, e.g
+              `@xkcd 1234`.
+            '''
+
+    def triage_message(self, message):
+        # Return True if we want to (possibly) response to this message
+        original_content = message['content']
+        is_xkcd_called_pattern = re.compile(r'^(@xkcd$|@xkcd(\s+))')
```

Comment:
Actually if we use `startswith()`, it will catch another word like `@xkcd123` or `@xkcdabc`